src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinSpawnerAnimals_optimizeSpawning.java

```diff
@@ -0,0 +1,319 @@
+                                        }
+
+                                        EntityLiving entityLiving;
+
+                                        try {
+                                            entityLiving = ((SpawnListEntryExt) spawnListEntry).constructEntity(world);
+                                        } catch (Exception exception) {
+                                            exception.printStackTrace();
+                                            return totalSpawnedEntities;
+                                        }
+
+                                        entityLiving.setLocationAndAngles(
+                                                spawnPosX,
+                                                spawnPosY,
+                                                spawnPosZ,
+                                                world.rand.nextFloat() * 360.0F,
+                                                0.0F);
+
+                                        Event.Result canSpawn = ForgeEventFactory
+                                                .canEntitySpawn(entityLiving, world, spawnPosX, spawnPosY, spawnPosZ);
+                                        if (canSpawn == Event.Result.ALLOW || (canSpawn == Event.Result.DEFAULT
+                                                && entityLiving.getCanSpawnHere())) {
+                                            ++packSize;
+                                            world.spawnEntityInWorld(entityLiving);
+                                            if (!ForgeEventFactory.doSpecialSpawn(
+                                                    entityLiving,
+                                                    world,
+                                                    spawnPosX,
+                                                    spawnPosY,
+                                                    spawnPosZ)) {
+                                                entityLivingData = entityLiving.onSpawnWithEgg(entityLivingData);
+                                            }
+
+                                            if (spawnAttempt >= ForgeEventFactory.getMaxSpawnPackSize(entityLiving)) {
+                                                break;
+                                            }
+                                        }
+                                        totalSpawnedEntities += packSize;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return totalSpawnedEntities;
+    }
+
+    /**
+     * Returns whether or not the specified creature type can spawn at the specified location.
+     *
+     * @author mitchej123
+     * @reason Optimize mob spawning
+     */
+    @Overwrite
+    public static boolean canCreatureTypeSpawnAtLocation(EnumCreatureType creatureType, World world, int x, int y,
```

Comment:
Currently small optimizations in short circuiting to avoid calling more expensive methods, and avoid calling getBlock() more than needed.

Main reason for overwriting is to be able to add server owner limits/controls on various mob types.